libsrc/cx16/waitvsync.s

```diff
@@ -5,17 +5,20 @@
 ; /* Wait for the start of the next video field. */
 ;
 ; VERA's vertical sync causes IRQs which increment the jiffy timer.
+;
+; Updated by ZeroByteOrg to use Kernal API RDTIM to retreive the TIMER variable
 ;
 
         .export         _waitvsync
 
-        .include        "cx16.inc"
-
-_waitvsync:
-        ldx     RAM_BANK        ; (TIMER is in RAM bank 0)
-        stz     RAM_BANK
-        lda     TIMER + 2
-:       cmp     TIMER + 2
-        beq     :-              ; Wait for next jiffy
-        stx     RAM_BANK
-        rts
+.proc _waitvsync: near
+      RDTIM = $FFDE  ; Kernal API for reading the jiffy timer
+      jsr RDTIM
+      sta lastjiffy
+keep_waiting:
+      jsr RDTIM
+      cmp #$FF       ; self-mod the value returned by RDTIM to save memory
```

Comment:
Self-modifying code must be put into the `.data` segment (the `.code` segment might be put into ROM).

Reply:
Indeed, `.importzp tmp1` is even shorter.

Reply:
Yes, please use tmp1 instead, no point in using selfmod here

Reply:
alright - I'll do that and update the PR. Makes sense for the ROM thing.